cfme/tests/containers/test_views_objects.py

```diff
@@ -0,0 +1,73 @@
+import pytest
+from cfme.fixtures import pytest_selenium as sel
+from cfme.web_ui import toolbar as tb
+from utils import testgen
+from utils.version import current_version
+
+
+pytestmark = [
+    pytest.mark.uncollectif(
+        lambda: current_version() < "5.5"),
+    pytest.mark.usefixtures('setup_provider')]
+pytest_generate_tests = testgen.generate(
+    testgen.container_providers, scope="function")
+
+
+def test_pods_views():
+    sel.force_navigate('containers_pods')
+    tb.select('Grid View')
+    assert tb.is_active('Grid View'), "Pods grid view setting failed"
```

Comment:
An example here (I ran the collection in pytest so it should be working):

```python
from itertools import product

LOCATIONS = ['containers_pods', 'containers_replicators']
VIEWS = ['Grid View', 'Tile View', 'List View']


@pytest.mark.parametrize(('location', 'view'), product(LOCATIONS, VIEWS))
def test_view_setting_whatever_you_like(location, view):
    """This tests checks that setting of the display mode has the desired effect

    Prerequisities:
        * Have a container provider set up.

    Steps:
        * Go to the desired location
        * Set X
        * Verify X
    """
    sel.force_navigate(location)
    tb.select(view)
    assert tb.is_active(view), "{}'s {} setting failed".format(location, view)
```

The resulting collection is:
```
  <Function 'test_view_setting_whatever_you_like[containers_pods-Grid View]'>
  <Function 'test_view_setting_whatever_you_like[containers_pods-Tile View]'>
  <Function 'test_view_setting_whatever_you_like[containers_pods-List View]'>
  <Function 'test_view_setting_whatever_you_like[containers_replicators-Grid View]'>
  <Function 'test_view_setting_whatever_you_like[containers_replicators-Tile View]'>
  <Function 'test_view_setting_whatever_you_like[containers_replicators-List View]'>
```

By the way I guess this should use `setup_a_provider` rather than testgen. [Here](https://github.com/ManageIQ/integration_tests/blob/master/cfme/tests/cloud_infra_common/test_tag_objects.py#L12) is an example. Your tests do not use the `provider` fixture therefore the provider does not get set up. You need to write the fixture that sets up the provider, can be module scoped, then you put the fixture name in the usefixtures.